src/types.ts

```diff
@@ -177,6 +178,15 @@ export interface ExcalidrawProps {
     appState: AppState,
     canvas: HTMLCanvasElement | null,
   ) => void;
+  /**
+   * Paste from clipboard.
+   *
+   * Returns `true` to prevent default.
+   */
+  onPasteFromClipboard?: (
+    data: Awaited<ReturnType<typeof parseClipboard>>,
+    event: ClipboardEvent | null,
+  ) => Promise<boolean> | boolean;
```

Comment:
lets rename to `onPaste`
Also the return type can be simply boolean, any specific use case for promise ?
The `data` is already resolved on prev line before triggering this callback so the type can be simply updated to the resolved data instead, though I would prefer exporting `ClipboardData` from `clipboard.ts`
```js
  onPaste?: (
    data: {
      spreadsheet?: Spreadsheet;
      elements?: readonly ExcalidrawElement[];
      text?: string;
      errorMessage?: string;
    },
    event: ClipboardEvent | null,
  ) => boolean;
```

Reply:
1. I do have a specific need for the `Promise<boolean>`… I need to a http request to know if the `id` is valid…

2. And `data` is resolved:
![image](https://user-images.githubusercontent.com/1761469/114160824-9f47f700-9927-11eb-82fe-938aa73e62fa.png)

The `Awaited<ReturnType<typeof parseClipboard>>` is to inferred the resolved type from  the `parseClipboard` function itself (so this is always in sync), but I can export a simpler type if you prefer.

3. Let's rename to `onPaste` 👍